### PR TITLE
Refactor: `MainTrace` type

### PR DESCRIPTION
**Pull request by iammadab**
## Describe your changes
This PR adds the type alias `MainTrace` for ColMatrix<Felt>.
relevant issue: #1201

### CODE REVIEW

File: processor/src/trace/mod.rs [hunk 1]
```diff
@@ -45,6 +45,9 @@ pub struct AuxTraceHints {
     pub(crate) chiplets: ChipletsAuxTraceBuilder,
 }
 
+// TODO: add documentation
+pub type MainTrace = ColMatrix<Felt>;
```
Comment (plafer) [inline processor/src/trace/mod.rs:49]: The `MainTrace` that the issue was referring to was [this one](https://github.com/0xPolygonMiden/miden-vm/blob/2d417030b7d248991a3a1431510f73fd8f59d843/air/src/trace/main_trace.rs#L31-L33) (it's on the `al-simplify-aux` branch)

**Review by plafer — CHANGES_REQUESTED** [review]
Thank you for the contribution!

The issue was actually referring to work happening on the branch `al-simplify-aux`, where the `MainTrace` was defined (see other comment in the review). I apologize for not making this clearer in the issue itself.

If you're willing to, you would need to

1. work off the `al-simplify-aux` branch
2. Do basically what you did here, except on that branch, using the `MainTrace` that's already defined.
3. Open another PR that targets the `al-simplify-aux` branch

Since `al-simplify-aux` is quite a bit different from `next`, it would probably be easier to close this PR, and start over on that branch. Let me know if anything needs clarification!

File: processor/src/trace/mod.rs (hunk 1 above)
Comment (iammadab) [reply processor/src/trace/mod.rs:49]: got it thanks

**Comment by iammadab** [comment]
@plafer closing this, see #1214.

**Pull request closed without merging**